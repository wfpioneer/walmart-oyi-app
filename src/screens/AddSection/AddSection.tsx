import React, { useEffect } from 'react';
import {
  FlatList,
  Keyboard,
  KeyboardAvoidingView,
  Text,
  View
} from 'react-native';
import { Dispatch } from 'redux';
import { useDispatch } from 'react-redux';
import { Header } from '@react-navigation/stack';
import { NavigationProp, useNavigation } from '@react-navigation/native';
import styles from './addSection.style';
import { strings } from '../../locales';
import { useTypedSelector } from '../../state/reducers/RootReducer';
import { CreateAisles, LocationIdName } from '../../state/reducers/Location';
import NumericSelector from '../../components/NumericSelector/NumericSelector';
import Button from '../../components/buttons/Button';
import { CREATE_FLOW } from '../../models/LocationItems';
import { setAisleSectionCount } from '../../state/actions/Location';
import { postCreateAisles } from '../../state/actions/saga';
import { hideActivityModal, showActivityModal } from '../../state/actions/Modal';
import { CreateAisleRequest, CreateAisleResponse } from '../../models/CreateZoneAisleSection.d';
import { AsyncState } from '../../models/AsyncState';
import { showSnackBar } from '../../state/actions/SnackBar';

interface AddSectionProps {
  aislesToCreate: CreateAisles[];
  selectedZone: { id: number; name: string };
  newZone: string;
  createFlow: CREATE_FLOW;
  dispatch: Dispatch<any>;
  navigation: NavigationProp<any>;
  existingSections: number;
  currentAisle: LocationIdName;
  modal: any;
  createAislesApi: AsyncState;
}

interface RenderAisles {
  item: CreateAisles;
  index: number;
}

const NEW_SECTION_MAX = 99;
const SECTION_MIN = 1;

const validateNumericInput = (sections: number, existingSections = 0): boolean => (sections >= SECTION_MIN)
  && (sections <= (NEW_SECTION_MAX - existingSections));

const validateSectionCounts = (aislesToCreate: CreateAisles[], existingSections: number): boolean => {
  let validation = true;
  aislesToCreate.forEach(aisle => {
    if ((aisle.sectionCount < SECTION_MIN) || (aisle.sectionCount > (NEW_SECTION_MAX - existingSections))) {
      validation = false;
    }
  });
  return validation;
};

export const AddSectionScreen = (props: AddSectionProps): JSX.Element => {
  const { createAislesApi, dispatch, navigation } = props;

  const navigateBack = () => {
    dispatch({ type: 'API/POST_CREATE_AISLES/RESET' });
    navigation.navigate('Aisles');
  };

  const handleAisleSectionCountIncrement = (aisleIndex: number, sectionCount: number) => {
    if (sectionCount < NEW_SECTION_MAX - props.existingSections) {
      props.dispatch(setAisleSectionCount(aisleIndex, sectionCount + 1));
    }
  };

  const handleAisleSectionCountDecrement = (aisleIndex: number, sectionCount: number) => {
    if (sectionCount > SECTION_MIN) {
      props.dispatch(setAisleSectionCount(aisleIndex, sectionCount - 1));
    }
  };

  const handleTextSectionCountChange = (text: string, aisleIndex: number) => {
    const newQty = parseInt(text, 10);
    if (!Number.isNaN(newQty)) {
      props.dispatch(setAisleSectionCount(aisleIndex, newQty));
    }
  };

  const renderAisles = (aisle: RenderAisles) => {
    const zoneName = props.createFlow === CREATE_FLOW.CREATE_ZONE ? props.newZone : props.selectedZone.name;
    const fullAisleName = `${strings('LOCATION.AISLE')} ${zoneName}${aisle.item.aisleName}`;
    return (
      <View style={styles.aisleContainer} key={fullAisleName}>
        <View style={styles.aisleNumericContainer}>
          <Text style={styles.aisleText}>
            {fullAisleName}
          </Text>
          <NumericSelector
            isValid={validateNumericInput(aisle.item.sectionCount, props.existingSections)}
            onDecreaseQty={() => {
              handleAisleSectionCountDecrement(aisle.index, aisle.item.sectionCount);
            }}
            onIncreaseQty={() => {
              handleAisleSectionCountIncrement(aisle.index, aisle.item.sectionCount);
            }}
            onTextChange={(text: string) => {
              handleTextSectionCountChange(text, aisle.index);
            }}
            value={aisle.item.sectionCount}
          />
        </View>
        {!validateNumericInput(aisle.item.sectionCount, props.existingSections) && (
          <Text style={styles.invalidLabel}>
            {strings('ITEM.OH_UPDATE_ERROR', {
              min: SECTION_MIN,
              max: NEW_SECTION_MAX - props.existingSections
            })}
          </Text>
        )}
      </View>
    );
  };

  const handleContinue = () => {
    console.log('continue pressed');
    switch (props.createFlow) {
      case CREATE_FLOW.CREATE_ZONE: {
        // TODO implement createZoneAisleSection
        const createZoneAisleSectionRequest = {
          zoneName: props.newZone,
          aisles: props.aislesToCreate
        };
        break;
      }
      case CREATE_FLOW.CREATE_AISLE: {
        const createAisleSectionRequest: CreateAisleRequest = {
          zoneId: props.selectedZone.id,
          aisles: props.aislesToCreate
        };
        dispatch(postCreateAisles({ aislesToCreate: createAisleSectionRequest }));
        break;
      }
      case CREATE_FLOW.CREATE_SECTION: {
        // TODO implement createSection
        const createSectionRequest = {
          aisles: [{
            aisleId: props.currentAisle.id,
            sectionCount: props.aislesToCreate[0].sectionCount
          }]
        };
        break;
      }
      default:
        break;
    }
  };

  const handleUnhandledTouches = () => {
    Keyboard.dismiss();
    return false;
  };

  // activity modal
  useEffect(() => {
    if (navigation.isFocused()) {
      if (!props.modal.showActivity) {
        if (createAislesApi.isWaiting) {
          dispatch(showActivityModal());
        }
      } else if (!createAislesApi.isWaiting) {
        dispatch(hideActivityModal());
      }
    }
  }, [props.modal, createAislesApi]);

  // Create Aisles and Sections API
  useEffect(() => {
    if (navigation.isFocused() && !createAislesApi.isWaiting) {
      // Success
      if (createAislesApi.result) {
        const createdAisles = createAislesApi.result.data as Array<CreateAisleResponse>;
        switch (createAislesApi.result.status) {
          case 200:
            dispatch(showSnackBar(
              strings('LOCATION.AISLES_ADDED').replace('{number}', createdAisles.length.toString()),
              2000
            ));
            navigateBack();
            break;
          case 207: {
            let succeeded = 0;
            createdAisles.forEach(aisle => {
              if (aisle.status === 200) {
                succeeded++;
              }
            });
            dispatch(showSnackBar(
              `${strings('LOCATION.INCOMPLETE_AISLES_ADDED').replace('{number}', succeeded.toString())}`
              + `\n${strings('LOCATION.INCOMPLETE_AISLES_PLEASE_CHECK')}`,
              3000
            ));
            break;
          }
          default:
        }
      }

      // Failure
      if (createAislesApi.error) {
        dispatch(showSnackBar(strings('LOCATION.ADD_AISLES_ERROR'), 2000));
      }
    }
  }, [createAislesApi, navigation]);

  return (
    <KeyboardAvoidingView
      style={styles.safeAreaView}
      behavior="height"
      keyboardVerticalOffset={110}
      onStartShouldSetResponder={handleUnhandledTouches}
    >
      <View style={styles.bodyContainer}>
        <FlatList
          data={props.aislesToCreate}
          renderItem={renderAisles}
          keyExtractor={item => item.aisleName.toString()}
          removeClippedSubviews={false}
        />
      </View>
      <View style={styles.buttonContainer}>
        <Button
          title={strings('GENERICS.CONTINUE')}
          style={styles.doneButton}
          onPress={handleContinue}
          disabled={!validateSectionCounts(props.aislesToCreate, props.existingSections)}
        />
      </View>
    </KeyboardAvoidingView>
  );
};

const AddSection = (): JSX.Element => {
  const aisles = useTypedSelector(state => state.Location.aisles);
  const currentAisle = useTypedSelector(state => state.Location.selectedAisle);
  const aislesToCreate = useTypedSelector(state => state.Location.aislesToCreate);
  const selectedZone = useTypedSelector(state => state.Location.selectedZone);
  const newZone = useTypedSelector(state => state.Location.newZone);
  const createFlow = useTypedSelector(state => state.Location.createFlow);
  const modal = useTypedSelector(state => state.modal);
  const createAislesApi = useTypedSelector(state => state.async.postCreateAisles);
  const dispatch = useDispatch();
  const navigation = useNavigation();

  let existingSections = 0;
  if (createFlow === CREATE_FLOW.CREATE_SECTION) {
    const zoneAisles = aisles.find(aisle => aisle.aisleId === currentAisle.id);
    if (zoneAisles) {
      existingSections = zoneAisles.sectionCount;
    }
  }

  return (
    <AddSectionScreen
      aislesToCreate={aislesToCreate}
      selectedZone={selectedZone}
      newZone={newZone}
      createFlow={createFlow}
      dispatch={dispatch}
      navigation={navigation}
      existingSections={existingSections}
      currentAisle={currentAisle}
      modal={modal}
      createAislesApi={createAislesApi}
    />
  );
};

export default AddSection;
