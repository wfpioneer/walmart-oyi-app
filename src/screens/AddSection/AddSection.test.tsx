import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AddSectionScreen } from './AddSection';
import { CREATE_FLOW } from '../../models/LocationItems';

describe('AddZoneScreen', () => {
  const aislesToCreate = [
    {
      name: 1,
      sectionCount: 1
    },
    {
      name: 2,
      sectionCount: 1
    },
    {
      name: 3,
      sectionCount: 1
    }
  ];
  const exiistingAisleToCreate = [
    {
      name: '1',
      sectionCount: 4
    }
  ];
  const aisleWithInvilidSectionCount = [
    {
      name: 1,
      sectionCount: 100
    }
  ];
  const currentZone = {
    id: 1,
    name: 'A'
  };
  const currentAisle = {
    id: 2,
    name: '1'
  };
  const defaultExistingSections = 0;
  const selectedZone = 'A';
  const renderer = ShallowRenderer.createRenderer();
  it('AddSectionScreen with valid input from add zone', () => {
    renderer.render(<AddSectionScreen
      aislesToCreate={aislesToCreate}
      selectedZone={currentZone}
      newZone={selectedZone}
      createFlow={CREATE_FLOW.CREATE_ZONE}
      dispatch={jest.fn()}
      existingSections={defaultExistingSections}
      currentAisle={currentAisle}
    />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('AddSectionScreen with valid input from add aisle', () => {
    renderer.render(<AddSectionScreen
      aislesToCreate={aislesToCreate}
      selectedZone={currentZone}
      newZone={selectedZone}
      createFlow={CREATE_FLOW.CREATE_AISLE}
      dispatch={jest.fn()}
      existingSections={defaultExistingSections}
      currentAisle={currentAisle}
    />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('AddSectionScreen with valid input from add section', () => {
    renderer.render(<AddSectionScreen
      aislesToCreate={exiistingAisleToCreate}
      selectedZone={currentZone}
      newZone={selectedZone}
      createFlow={CREATE_FLOW.CREATE_SECTION}
      dispatch={jest.fn()}
      existingSections={defaultExistingSections}
      currentAisle={currentAisle}
    />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('AddSectionScreen with invalid section count', () => {
    renderer.render(<AddSectionScreen
      aislesToCreate={aisleWithInvilidSectionCount}
      selectedZone={currentZone}
      newZone={selectedZone}
      createFlow={CREATE_FLOW.CREATE_ZONE}
      dispatch={jest.fn()}
      existingSections={defaultExistingSections}
      currentAisle={currentAisle}
    />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
