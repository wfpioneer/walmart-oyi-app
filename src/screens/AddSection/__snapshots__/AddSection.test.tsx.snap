// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSection Screen render tests AddSectionScreen with invalid section count 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 100,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests AddSectionScreen with valid input from add aisle 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 1,
          },
          Object {
            "aisleName": 2,
            "sectionCount": 1,
          },
          Object {
            "aisleName": 3,
            "sectionCount": 1,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests AddSectionScreen with valid input from add section 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": "1",
            "sectionCount": 4,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests AddSectionScreen with valid input from add zone 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 1,
          },
          Object {
            "aisleName": 2,
            "sectionCount": 1,
          },
          Object {
            "aisleName": 3,
            "sectionCount": 1,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests Waits on the create zone/aisles/sections api 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 100,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests addSectionScreen after createSections api called and failed 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": "1",
            "sectionCount": 4,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests addSectionScreen after createSections api called and successful 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": "1",
            "sectionCount": 4,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests addSectionScreen while waiting for api to finish 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": "1",
            "sectionCount": 4,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={false}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests failed on the create aisles/sections api 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 100,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests partially succeeded on the create aisles/sections api 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 100,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests partially succeeded on the create aisles/sections api 2`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 100,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests succeeded on the create aisles/sections api 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 100,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests succeeded on the create zone/aisles/sections api 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 100,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;

exports[`AddSection Screen render tests waits on the create aisles/sections api 1`] = `
<KeyboardAvoidingView
  behavior="height"
  enabled={true}
  keyboardVerticalOffset={110}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "flex": 0.9,
        "justifyContent": "flex-start",
        "marginTop": 5,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      data={
        Array [
          Object {
            "aisleName": 1,
            "sectionCount": 100,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginBottom": 15,
      }
    }
  >
    <Button
      backgroundColor="#0071e9"
      disabled={true}
      height={45}
      onPress={[Function]}
      radius={5}
      style={
        Object {
          "margin": 15,
        }
      }
      title="[missing \\"en.GENERICS.SUBMIT\\" translation]"
      titleColor="#ffffffff"
      titleFontSize={16}
      titleFontWeight="normal"
      type={1}
      width={-1}
    />
  </View>
</KeyboardAvoidingView>
`;
